io/io/src/TStreamerInfo.cxx

```diff
@@ -253,6 +253,62 @@ namespace {
       TStreamerInfo* fInfo;
    };
 
+   decltype(auto) GetSourceType(ROOT::TSchemaRule::TSources *s)
```

Comment:
Why do we need `decltype(auto)` instead of the spelled out type `std::pair<...>`?

Reply:
To avoid redundant typing of the types.  For example, this function started with 2 return values (in this commit )  and grew to 5 later on without having to change this line.  Since this in an internal routine, I think it is fine to keep it opaque.